src/frontend/common.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import moment from 'moment-timezone';
+import { isPresent } from 'ts-is-present';
+import { displayText, FORMAT_DEFAULT_AND_ORIGINAL, FORMAT_DEFAULT_AND_UTC } from "../displayOptions";
+import { Lozenge, Tooltip, Text } from '@forge/react';
+
+/**
+ * Validates the config object and returns error message if missing.
+ */
+export function validateConfig(details, config, requiredFields = ['date', 'time', 'timeZone']) {
+  if (!isPresent(details)) return "Macro not configured. Please configure it in the Macro Configuration.";
+  if (!isPresent(config)) return "Macro not configured. Please configure it in the Macro Configuration.";
+  for (const field of requiredFields) {
+    if (!isPresent(config[field])) return `${field.charAt(0).toUpperCase() + field.slice(1)} not configured. Please configure it in the Macro Configuration.`;
+  }
+  return null;
+}
+
+/**
+ * Parses and validates time string. Returns { time, meridiem } or error message.
+ */
+export function parseTime(rawTime) {
+  const timeMatch = /^(?<time>(1[012]|0?[1-9]):[0-5][0-9])\s?(?<meridiem>[aApP][mM])$/;
+  const matchResult = timeMatch.exec(rawTime);
+  if (!matchResult) {
+    return { error: `Time "${rawTime}" not in the "XX:XX (am|pm)" format! Example, valid values are: "9:00 am" or "12:45 pm".` };
+  }
+  return matchResult.groups;
+}
+
+export function renderDateLozenge({
```

Comment:
You could argue that this belongs in displayOptions.js, but because it is a rendering function I've decided to keep it away from the logic functions.